polyply/src/minimizer.py

```diff
@@ -77,8 +77,15 @@ def compute_dih(params, coords):
     -------
     float
     """
-    dih_angle = dih(coords[0], coords[1], coords[2], coords[3])
-    return (dih_angle - float(params[1]))**2.0
+    # only optimize imporper dihedrals
+    # we filter all non-imporpers out here because
+    # the itp file parser doesn't distinguish them
```

Comment:
This is missing the "why". Also, don't improper type 4 also have multiplicity?

Reply:
good point but it's hard to tell what it does. I can't think of a use case at the moment